azurerm/resource_arm_iothub_dps_shared_access_policy.go

```diff
@@ -217,6 +234,14 @@ func resourceArmIotHubDPSSharedAccessPolicyRead(d *schema.ResourceData, meta int
 	d.Set("primary_key", accessPolicy.PrimaryKey)
 	d.Set("secondary_key", accessPolicy.SecondaryKey)
 
+	if err := d.Set("primary_connection_string", getSAPConnectionString(*iothubDps.Properties.ServiceOperationsHostName, keyName, *accessPolicy.PrimaryKey)); err != nil {
```

Comment:
We'll need to nil check these properties here:
```suggestion
	if props := iothubDps.Properties; props != nil {
		if hosts := props.ServiceOperationsHostName; hosts != nil
			if pkey = accessPolicy.PrimaryKey; pkey != nil {
				if err := d.Set("primary_connection_string", getSAPConnectionString(*iothubDps.Properties.ServiceOperationsHostName, keyName, *accessPolicy.PrimaryKey)); err != nil {
```